spec/models/user_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+describe User do
+  let(:user){ build(:user) }
+
+  # Happy
+  it "is valid with default attributes" do
+    expect(user).to be_valid
+  end
+
+  it "has password length bigger then 4" do
+    user.password = "1"
+    expect(user).to_not be_valid
+  end
+
+  it "has password length smaller then 24" do
+    user.password = "1"*25
+    expect(user).to_not be_valid    
+  end
+
+  it "has email with @ sign" do
```

Comment:
Didn't think of this one as it was automatically done by one of the gems, good thinking! I'll have to add as well.